File: sequencer.rb
class Sequence
  def go(initial_value, iterations)
    current_value = initial_value.to_s
    iterations.times do 
      array_result = string_to_formatted_array(current_value)
      current_value = array_result.map{|r| "#{r.length}#{r.first}"}.to_s
      print current_value
      print "\n"
    end
  end

  private

  def string_to_formatted_array(input)
    final_array=Array.new
    char_split = input.to_s.split''

    char_split.each_with_index do |char, index|
      if index > 0 && char_split[index] == char_split[index-1]
        final_array.last << char
      else
        final_array << [char]
      end
      
    end
    final_array
  end

end

new_sequencer = Sequence.new
new_sequencer.go(1,10)

File: sequencer.js
var Sequencer = function(){
  this.go = function(initial_value, iterations){
    var current_value = initial_value;
    var array_result;
    for (var i=0; i<iterations; i++){
      array_result = this.string_to_formatted_array(current_value);
      current_value = "";
      array_result.map( function(item) { current_value += item.length.toString() + item[0]; } );
      console.log(current_value);
    };
  };

};

Sequencer.prototype.string_to_formatted_array = function(the_input){
  var result = new Array();
  var input_string = the_input.toString();
  for (var i=0; i<input_string.length; i++){
    if ((i>0) && (input_string[i] == input_string[i-1])){
      result[result.length-1].push(input_string[i]);
    }else {
      result.push(new Array(input_string[i]));
    };
  };
  return result; 
};

var new_sequencer = new Sequencer();
new_sequencer.go(1,10);
